### ISSUE TITLE
Release 10.0

### ISSUE THREAD

**Issue by sunshinejr** [Moya/Moya#1253]
After 9.0 release, ~~which is right around the corner~~ which is already done, we should start thinking about the next release. Current list of tasks:

- [x] Fully working Swift 4.0 support (means upgrading RxSwift/ReactiveSwift versions)
- [ ] ~Sample data refractor (see issue #1126 and discussion #1285). We could think about availability of an external plugin (out of the core) that saves stubs (if we can do something to let people create this one)~ **Delayed**
- [x] Propagate parameter encoding error for URLRequest mapping (PR: #1248)
- [x] Remove default parameters from Endpoint (see issue #1247, PR: #1289)
- [x] Update `NetworkActivityPlugin` to pass `Target` in the `networkActivityClosure` (see issue #1184, PR: #1290)
- [x] Add Moya.Response extension for `Decodable` types (see issue #1118, PR: #1335)
- [x] Add `Task` case to handle request with `Encodable` type (see issue: #1325, PR: #1349)
- [x] SPM PackageDescription API migration from v3 to v4 (see issue #1240, PR: #1266)
- [x] Remove all deprecatations from 9.0.0 (PR: #1320)
- [x] Wait for a stable release of RxSwift
- [ ] Wait for a stable release of ReactiveSwift

**Comment by SD10** [comment]
We still have a good amount of time to work on `10.0` but I think it's important to note that the `RxSwift`/ `ReactiveSwift` support should be the top priority. We can always iterate quickly by shipping `10.0` and then `11.0` soon after.

The reason I say this is because I've given a lot of thought to the sample data refactor (#1126) and IMO I only see two *good* solutions. I think it's important that the initial PR for this receives input from a handful of Moya contributors. 

Essentially, let's not rush any core API decisions to get full Swift 4.0 / reactive support 😃

**Comment by sunshinejr** [comment]
Sure! This list is rather something we could _potentially_ do, so few of them might not get into the list of features in 10.0. I agree that we should release the next version around the time Swift 4.0 comes out officially.

**Comment by sunshinejr** [comment]
Hey all, I've also created a branch that is completely Swift 4.0.0-compatible! Branch name is `10.0.0-dev`. So install step for CocoaPods would be one of the following:
```rbi
pod 'Moya/RxSwift', :git => 'https://github.com/Moya/Moya.git', :branch => '10.0.0-dev'
# or
pod 'Moya/ReactiveSwift', :git => 'https://github.com/Moya/Moya.git', :branch => '10.0.0-dev'
```

**Comment by sunshinejr** [comment]
Hey all! I'm thinking that after Decodable/Encodable extensions and removing deprecated methods we could release a beta. I'm not sure we can get the sample data refactor in the 10.0 as I know @SD10 had some troubles with that one and it might be a lot bigger than expected. What do you guys think?

**Comment by SD10** [comment]
Yes, let's delay the sample data refactor again and move it to Moya `11.0`. It's worth taking the time to find a long-term solution and one that will play well with Harvey. In #1285, I suggest a short-term solution to keeping `sampleData` in the Test Target for those who are concerned.

**Comment by evermeer** [comment]
Please publish 10.0 as soon as possible. I have a cocoapod with a dependency on Moya and RxSwift and I'm not able to publish my Swift 4 version. 

https://github.com/evermeer/EVReflection/issues/236

**Comment by SD10** [comment]
Hey @evermeer I'm really sorry for the inconvenience. We're trying to get a 10.0 beta out ASAP. I'll ping you with more updates as I have them

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
It looks like it could be difficult to find the right solution for #1325 in a timely manner. Does anyone this we should delay this until the next version? It would be nice to group more breaking changes into this release but we probably want to cut an official beta release with Swift 4 support by now

**Comment by sunshinejr** [comment]
@SD10 there were some troubles, but the PR is there: #1349. I found a way to make the API call a little more user-friendly without casting to `AnyEncodable` and it seems to be working fine. After that one we are ready to 🚢 beta version! 🎉

**Comment by sunshinejr** [comment]
Alrighty! Today, after #1349 & #1350 are merged, we should aim for releasing 10.0.0-beta.1!

**Comment by sunshinejr** [comment]
10.0.0-beta.1 released! 🎉 Just before the release RxSwift `4.0.0-rc.0` just came out so I've updated all package managers to cover it as well. Thus, to use `Moya/RxSwift` you need `4.0.0-rc.0` version of `RxSwift` as well. For CocoaPods, it means:

```ruby
pod 'RxSwift', '4.0.0-rc.0'
pod 'Moya/RxSwift', '10.0.0-beta.1'
```

**Comment by SD10** [comment]
Congratulations everyone 💪 🎉 Thank you for moving this forward @sunshinejr 😃!
cc @evermeer

**Comment by BasThomas** [comment]
@sunshinejr is there anything we still need to do for `10.0` except waiting for stable `RxSwift` and `ReactiveSwift` releases?

**Comment by sunshinejr** [comment]
Fix bugs if there are any! 😄 But other than that we are good to go @BasThomas :)

**Comment by sunshinejr** [comment]
Okay guys, the bug fix is there, I'm gonna try to push a release in the next few hours or so 😉

**Comment by sunshinejr** [comment]
Moya 10.0.0 released! You can see the release [here](https://github.com/Moya/Moya/releases/tag/10.0.0). Big ❤️ to everyone involved:

@devxoul @SD10 @LeLuckyVint @afonsograca @AndrewSB @BasThomas @pedrovereza @sunshinejr (sorry if I forgot about someone)

🎉 🎉 🎉 

_I'm closing this one, but every problem/question about 10.0.0 let's just create new issue 😉_
